source/common/tracing/zipkin/zipkin_tracer_impl.cc

```diff
@@ -107,6 +108,20 @@ Tracing::SpanPtr Driver::startSpan(const Tracing::Config& config, Http::HeaderMa
 
     new_zipkin_span =
         tracer.startSpan(config, request_headers.Host()->value().c_str(), start_time, context);
+  } else if (request_headers.XB3TraceId() && request_headers.XB3SpanId()) {
+    uint64_t traceId(0);
+    uint64_t spanId(0);
+    uint64_t parentId(0);
+    if (!StringUtil::atoul(request_headers.XB3TraceId()->value().c_str(), traceId, 16)
+        || !StringUtil::atoul(request_headers.XB3SpanId()->value().c_str(), spanId, 16)
+        || (request_headers.XB3ParentSpanId() && !StringUtil::atoul(request_headers.XB3ParentSpanId()->value().c_str(), parentId, 16))) {
+      return nullptr;
```

Comment:
i'd probably return nullspan here to be on a safe side.

Reply:
Sorry yes I meant NullSpan (not nullptr)

Reply:
No problem will sort it tomorrow.